examples/shader/Shader.cpp

```diff
@@ -38,26 +38,29 @@ class Pixelate : public Effect
             return false;
         m_sprite.emplace(m_texture);
 
-        m_shader.setUniform("texture", sf::Shader::CurrentTexture);
+        // Load the shader
+        if (!(m_shader = sf::Shader::loadFromFile("resources/pixelate.frag", sf::Shader::Type::Fragment)))
+            return false;
+        m_shader->setUniform("texture", sf::Shader::CurrentTexture);
 
         return true;
     }
 
     void onUpdate(float, float x, float y) override
     {
-        m_shader.setUniform("pixel_threshold", (x + y) / 30);
+        m_shader->setUniform("pixel_threshold", (x + y) / 30);
     }
 
     void onDraw(sf::RenderTarget& target, sf::RenderStates states) const override
     {
-        states.shader = &m_shader;
+        states.shader = &*m_shader;
```

Comment:
I'm not sure about using `&*` to get a raw pointer here (and in the other instances), at least not in a tutorial/example context. For someone rather new this might feel confusing and unnecessary as you don't necessarily consider the overloaded operator in `std::optional`. Maybe use `&m_shader.value()` after all?